### PR TITLE
Rename Node3D's property translation to position

### LINKED ISSUE #1557: Change name of ' Translation ' to ' Position' in Transforms . .
**Describe the project you are working on:**
A Tomb Raider game . .

**Describe the problem or limitation you are having in your project:**
I was sitting making some visual script, and I noticed that the ' place ' of, a character, or object is called ' translation ' . . However, when making the ' logic ', for instance, to make a vector, I thought it makes more sense to say, ' GET position ' of enemy MINUS ' GET position ' of player, to make a vector, from player, to enemy . . I know ' translation ' is a pretty good name, but when we do the math, what we want to do, ' when thinking ', is GET position of something, MINUS position of something else . . I think under transforms, it might make more sense to write ' Position ', also since translation of refers to movement, ie. translate something from A to B, and not ' technically ' a position, it'd make more sense to call ' Translation ' ' Position ' . .

That way, when we make boxes in visual script, the names of the boxes make more sense, or so . . ie. to make a vector from player to enemy the box, should be called Get Position ( enemy ) MINUS Get Position ( Player ), not ' get translation ' . . It's a little thing, but it'd make the engine have names, that ' match ' the words, for how we think, more . .

I was also thinking, since under Transform is says Rotation Degrees, same way some boxes have a ' Index ' setting, ie. for a rotation, one can disable ' Default Settings ', that one could have an ' index ' for a GET rotation box, that outputs either radians, or degrees . . I'm not sure, just some thoughts . . It'd make the engine a bit more, like what we ' think ', when doing math, that's all . . It'd just be a small help, especially for visual script coders, or so . .

Also, for some reason the written code, for keyboard presses is called Input(), but in visual script, it is called action . . imo, in visual script, it could be called Input also, or User_input, I think it makes more sense to make it match what is written code, since then after one has learned visual script, one is more ready, to begin learning written code, and so as much as possible, the words in visual script boxes match, what happens in written code, to help talk between written coders, and visual script game makers, or so . . These are very minor things, but they often bug me . .

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
These are just really small changes, that I think would make the coding experience a bit nicer, nothing more, a bit silly . . .

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Change transform name ' Translation ', to ' Position ', so when making code, it matches more, how we think in math, that's all . . .

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I can't re-change the engine, very well, as a new user, it'd take a long time . .

**Is there a reason why this should be core and not an add-on in the asset library?:**
Well, it would just make the engine make a bit more sense, to how we think, from math, making it a bit faster . .

### PR DESCRIPTION
**Pull request by madmiraal**
Currently, `Node3D` has a helper property called `translation`, which would be better called `position` as exists in `Node2D`. This PR renames `Node3D`'s `translation` property, `position`.
Part of #16863.

*Bugsquad edit:* Fixes https://github.com/godotengine/godot-proposals/issues/1557

### CODE REVIEW

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: doc/classes/Node3D.xml
Comment (madmiraal) [reply doc/classes/Node3D.xml]: I slightly modified your suggestion to: "Local position or translation of this node relative to the parent. This is equivalent to [code]transform.origin[/code]."

**Review by aaronfranke — COMMENTED** [review]
I support this change and the code looks good. Would also be nice to update the docs for Node2D.position to match though.

**Review by aaronfranke — APPROVED** [review]

**Comment by bluenote10** [comment]
See my concerns in the proposal https://github.com/godotengine/godot-proposals/issues/1557.

**Comment by madmiraal** [comment]
Rebased following merge of #45879.

**Comment by madmiraal** [comment]
Updated documentation with `--doctool .`.

**Comment by akien-mga** [comment]
We discussed this in a PR preview meeting today, and it had been discussed on and off in chat in previous months too. With the contributors present, we finally reached a consensus that this rename might indeed be a good idea to make things consistent.

`translation` is more "accurate" for experienced 3D developers, but Godot aims at being user-friendly also to less savvy developers, and `position` is still a correct description of what this represents. So we prefer standardizing on `position` for both 2D and 3D than the other way around, assuming that `translation` in 2D would confuse beginners even more. Other issues with `translation` is the homonymy with text/language translation (localization), and the similarity with `transform` which also starts with `trans-`.

**Review by akien-mga — APPROVED** [review]
Good to merge after a rebase.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Calinou** [comment]
Note that once this PR is merged, most existing 3D scenes will break until we add property rename handlers.

**Comment by aaronfranke** [comment]
@Calinou Surprisingly I don't actually think this is a problem. Exactly zero of the scenes in the demo projects have `translation =` in them, because 3D nodes currently store a `Transform3D` for their transformations. The only cases where `translation` exists in the scene files is with animation tracks that animate the `translation` property.

**Comment by Calinou** [comment]
> @Calinou Surprisingly I don't actually think this is a problem. Exactly zero of the scenes in the demo projects have `translation =` in them, because 3D nodes currently store a `Transform3D` for their transformations. The only cases where `translation` exists in the scene files is with animation tracks that animate the `translation` property.

Indeed, I was mistaken as only `transform` is saved in scene files.

**Pull request merged**
